pkg/controller/helper.go

```diff
@@ -30,6 +31,13 @@ func (o IssuerOptions) ResourceNamespace(iss cmapi.GenericIssuer) string {
 	return ns
 }
 
+func (o IssuerOptions) ResourceNamespaceFromRef(ref cmmeta.ObjectReference, ns string) string {
+	if ns != "" {
+		return ns
+	}
+	return o.ClusterResourceNamespace
+}
```

Comment:
question: why is this added? We don't use `ref` anywhere in the function body as far as I can see?

Reply:
You are right, we can just use the namespace of the challenge. Updated this in https://github.com/cert-manager/cert-manager/pull/7285/commits/f62e5e10935ec16b52edaecc1355d6d3ac007cb7.